packages/react-components/react-combobox/src/components/Dropdown/useDropdown.tsx

```diff
@@ -0,0 +1,69 @@
+import * as React from 'react';
+import { ChevronDownRegular as ChevronDownIcon } from '@fluentui/react-icons';
+import { getPartitionedNativeProps, resolveShorthand } from '@fluentui/react-utilities';
+import { useComboboxBaseState } from '../../utils/useComboboxBaseState';
+import { useTriggerListboxSlots } from '../../utils/useTriggerListboxSlots';
+import { useComboboxPopup } from '../../utils/useComboboxPopup';
+import { Listbox } from '../Listbox/Listbox';
+import type { DropdownProps, DropdownState } from './Dropdown.types';
+
+/**
+ * Create the state required to render Dropdown.
+ *
+ * The returned state can be modified with hooks such as useDropdownStyles_unstable,
+ * before being passed to renderDropdown_unstable.
+ *
+ * @param props - props from this instance of Dropdown
+ * @param ref - reference to root HTMLElement of Dropdown
+ */
+export const useDropdown_unstable = (props: DropdownProps, ref: React.Ref<HTMLButtonElement>): DropdownState => {
+  const baseState = useComboboxBaseState(props);
+
+  const { primary: triggerNativeProps, root: rootNativeProps } = getPartitionedNativeProps({
+    props,
+    primarySlotTagName: 'button',
+    excludedPropNames: ['children'],
+  });
+
+  const triggerShorthand = resolveShorthand(props.button, {
+    required: true,
+    defaultProps: {
+      type: 'button',
+      children: baseState.value || props.placeholder,
+      ...triggerNativeProps,
+    },
+  });
+
+  const listboxShorthand = resolveShorthand(props.listbox, { required: true });
+
+  const [triggerWithPopup, listboxWithPopup] = useComboboxPopup(props, triggerShorthand, listboxShorthand);
+  const [triggerSlot, listboxSlot] = useTriggerListboxSlots(props, baseState, ref, triggerWithPopup, listboxWithPopup);
```

Comment:
As I mentioned in the other PR: probably these hooks should accept and return `state` to be consistent with other components.

Reply:
I actually tried doing this initially, but the problem is essentially with typing. The issue is that Combobox vs. Dropdown have different and incompatible primary slot types (button vs. input, with different children types), and I didn't find a way to type the state prop and return type in a way that would work without a bunch of explicit `as X` casting.

I'd be happy to work with you on that more, if you have any thoughts on how to make that work. For now, both hooks are fully internal and not exported, so we can change this as much as we want without affecting the public API :)